### PR TITLE
Mock assembly for testing and its program generator

### PR DESCRIPTION
**Pull request by sklam**

### CODE REVIEW

File: numba_rvsdg/mock_asm.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        VM(buf).run(instlist)
+        got = buf.getvalue().split()
+
+    expected = ["Start", *(["A"] * 10), "B"]
+    assert got == expected
+
+
+def test_double_exchange_loop():
+    asm = """
+    print Start
```
Comment (esc) [inline numba_rvsdg/mock_asm.py:232]: I would recommend using the `textwrap.dedent` function here such that the `asm` is indented like the Python code that defines it.

Comment (sklam) [reply numba_rvsdg/mock_asm.py:232]: indentation shouldn't matter. i probably could have indent the asm to the same level to the python source

**Comment by sklam** [comment]
I'll be focusing on numba release this week. If you want to make changes to this PR or refactor it, feel free and don't wait for me.

File: numba_rvsdg/mock_asm.py (hunk 1 above)
Comment (esc) [reply numba_rvsdg/mock_asm.py]: done

**Comment by esc** [comment]
I had to split the tests and definition of the mock_asm module, perhaps the imports could be fixed up.

**Review by kc611 — APPROVED** [review]
LGTM!

**Pull request merged**